lib/postgrex/pgpass.ex

```diff
@@ -47,17 +36,45 @@ defmodule Postgrex.Pgpass do
     |> Enum.map(&(String.split(&1, ":")))
   end
 
-  defp readable_pgpass_file? do
-    with {:ok, stat} <- File.stat(pgpass_path()),
+  defp readable_pgpass_file?(nil) do
+    # when the pgpass is not passed via the :passfile option, simply ignore and move on
+    with default_path <- pgpass_path(),
+         {:ok, stat} <- File.stat(default_path),
          0o0600 <- stat.mode &&& 0o0777,
-    do: true,
+    do: {:ok, default_path},
     else: (_ -> false)
   end
+  defp readable_pgpass_file?(passfile) do
+    # when using the :passfile options, the file must be readable or an error is raised
+    case File.stat(passfile) do
```

Comment:
I rolled a custom exception because File.Error only wants posix codes and I want to be clear on what the problem is.